src/stores/Binding/Store.ts

```diff
@@ -658,15 +660,19 @@ const getInitialState = (
                                 undefined,
                             ];
 
-                    state.currentBinding = Boolean(
-                        evaluatedBindingUUID && evaluatedResourceConfig
-                    )
-                        ? {
-                              uuid: evaluatedBindingUUID,
-                              collection:
-                                  evaluatedResourceConfig.meta.collectionName,
-                          }
-                        : null;
+                    // Only update the current binding if we have removed the one that is currently selected
+                    if (state.currentBinding?.uuid === uuid) {
+                        state.currentBinding = Boolean(
+                            evaluatedBindingUUID && evaluatedResourceConfig
+                        )
+                            ? {
+                                  uuid: evaluatedBindingUUID,
+                                  collection:
+                                      evaluatedResourceConfig.meta
+                                          .collectionName,
+                              }
+                            : null;
+                    }
```

Comment:
We used to keep the current selected during removing... I think that works a bit better. So putting that back in.